src/Symfony/Component/Notifier/Bridge/GoIP/GoIPTransport.php

```diff
@@ -0,0 +1,120 @@
+        EventDispatcherInterface $dispatcher = null
+    ) {
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('goip://%s?sim_slot=%s', $this->getEndpoint(), $this->simSlot);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage && (null === $message->getOptions() || $message->getOptions() instanceof GoIPOptions);
+    }
+
+    /**
+     * @throws TransportExceptionInterface
+     * @throws ServerExceptionInterface
+     * @throws RedirectionExceptionInterface
+     * @throws ClientExceptionInterface
+     */
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        if (($options = $message->getOptions()) && !$options instanceof GoIPOptions) {
+            throw new LogicException(sprintf('The "%s" transport only supports an instance of the "%s" as an option class.', __CLASS__, GoIPOptions::class));
+        }
+
+        if ('' !== $message->getFrom()) {
+            throw new LogicException(sprintf('The "%s" transport does not support the "From" option.', __CLASS__));
+        }
+
+        $response = $this->client->request('GET', $this->getEndpoint(), [
+            'query' => [
+                'u' => $this->username,
+                'p' => $this->password,
+                'l' => $options->getSIMSlot() ?? $this->simSlot,
+                'n' => $message->getPhone(),
+                'm' => $message->getSubject(),
+            ],
+        ]);
+
+        try {
+            $statusCode = $response->getStatusCode();
+        } catch (TransportExceptionInterface $e) {
+            throw new TransportException('Could not reach the GoIP gateway.', $response, 0, $e);
+        }
+
+        if (200 !== $statusCode) {
+            throw new TransportException(sprintf('The GoIP gateway did not respond correctly on the address: "%s", HTTP-Status-Code: "%s" .', $this->getEndpoint(), $statusCode), $response);
+        }
+
+        if (!str_contains(strtolower($response->getContent()), 'sending') || str_contains(strtolower($response->getContent()), 'error')) {
+            throw new TransportException(sprintf('Could not send the message through GoIP. Response: "%s".', $response->getContent()), $response);
+        }
+
+        if (!$messageId = $this->extractMessageIdFromContent($response->getContent())) {
+            throw new TransportException(sprintf('Could not extract the message id from the GoIP response: "%s".', $response->getContent()), $response);
+        }
+
+        $sentMessage = new SentMessage($message, (string) $this);
+        $sentMessage->setMessageId($messageId);
+
+        return $sentMessage;
+    }
+
+    private function extractMessageIdFromContent(string $content): string|bool
```

Comment:
can you please add some tests covering both example cases?

I would also move the comment above the method

Reply:
Done